cumulus/parachains/runtimes/assets/common/src/matching.rs

```diff
@@ -102,19 +102,27 @@ impl<
 pub struct RemoteAssetFromLocation<AssetsAllowedNetworks, OriginLocation>(
 	core::marker::PhantomData<(AssetsAllowedNetworks, OriginLocation)>,
 );
-impl<AssetsAllowedNetworks: Contains<Location>, OriginLocation: Get<Location>>
-	ContainsPair<Asset, Location> for RemoteAssetFromLocation<AssetsAllowedNetworks, OriginLocation>
+impl<
+		L: TryInto<Location> + Clone,
```

Comment:
maybe `xcm::TryAs` would also work here:
```suggestion
		L: TryAs<Location>,
```

so we don't  need to clone anything here:
```
		let Ok(asset) = asset.try_as() else {
			return false;
		};
		let Ok(origin) = origin.try_as() else {
			return false;
		};
```

Reply:
the trait `TryAs<cumulus_primitives_core::Location>` is not implemented for `cumulus_primitives_core::Location`, which is required by `RemoteAssetFromLocation<AssetsAllowedNetworks, OriginLocation>: ContainsPair<cumulus_primitives_core::Location, cumulus_primitives_core::Location>

although it should work...